app/slashcommands/command_share.go

```diff
@@ -0,0 +1,296 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package slashcommands
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	goi18n "github.com/mattermost/go-i18n/i18n"
+
+	"github.com/mattermost/mattermost-server/v5/app"
+	"github.com/mattermost/mattermost-server/v5/model"
+)
+
+type ShareProvider struct {
+}
+
+const (
+	CommandTriggerShare   = "share"
+	AvailableShareActions = "Available actions: share_channel, unshare_channel, invite_remove, uninvite_remote, status"
+)
+
+func init() {
+	app.RegisterCommandProvider(&ShareProvider{})
+}
+
+func (sp *ShareProvider) GetTrigger() string {
+	return CommandTriggerShare
+}
+
+func (sp *ShareProvider) GetCommand(a *app.App, T goi18n.TranslateFunc) *model.Command {
+	share := model.NewAutocompleteData(CommandTriggerShare, "[action]", "Available commands: share_channel, unshare_channel, invite_remote, uninvite_remote, status")
+
+	shareChannel := model.NewAutocompleteData("share_channel", "", "Share the current channel")
+	shareChannel.AddNamedTextArgument("readonly", "Channel will be shared in read-only mode", "[readonly] - 'Y' or 'N'.  Defaults to 'N'", "Y|N|y|n", false)
+	shareChannel.AddNamedTextArgument("name", "Channel name provided to remote instances", "[name] - defaults to channel name", "", false)
+	shareChannel.AddNamedTextArgument("displayname", "Channel display name provided to remote instances", "[displayname] - defaults to channel displayname", "", false)
+	shareChannel.AddNamedTextArgument("purpose", "Channel purpose provided to remote instances", "[purpose] - defaults to channel purpose", "", false)
+	shareChannel.AddNamedTextArgument("header", "Channel header provided to remote instances", "[header] - defaults to channels header", "", false)
+
+	unshareChannel := model.NewAutocompleteData("unshare_channel", "", "Unshares the current channel")
+	unshareChannel.AddNamedTextArgument("are_you_sure", "Are you sure? This channel will be unshared and all remote instances will be uninvited", "'Y' or 'N'", "Y|N|y|n", true)
```

Comment:
nit: would `confirm` be more convenient?
```suggestion
	unshareChannel.AddNamedTextArgument("confirm", "Are you sure? This channel will be unshared and all remote instances will be uninvited", "'Y' or 'N'", "Y|N|y|n", true)
```

Reply:
Ordinarily yes, but convenience for this one is not desired.  It should be a bit of effort to do a destructive command like this.